nmt.go

```diff
@@ -171,27 +169,66 @@ func (n NamespacedMerkleTree) ProveNamespace(nID namespace.ID) (Proof, error) {
 	// the range it would be in (to generate a proof of absence and to return
 	// the corresponding leaf hashes).
 	n.computeLeafHashesIfNecessary()
-	subTreeHasher := internal.NewCachedSubtreeHasher(n.leafHashes, n.treeHasher)
-	var err error
-	proof, err := merkletree.BuildRangeProof(proofStart, proofEnd, subTreeHasher)
-	if err != nil {
-		// This should never happen.
-		// TODO would be good to back this by more tests and fuzzing.
-		return Proof{}, fmt.Errorf(
-			"unexpected err: %w on nID: %v, range: [%v, %v)",
-			err,
-			nID,
-			proofStart,
-			proofEnd,
-		)
-	}
+	proof := n.buildRangeProof(proofStart, proofEnd)
 
 	if found {
 		return NewInclusionProof(proofStart, proofEnd, proof, isMaxNsIgnored), nil
 	}
 	return NewAbsenceProof(proofStart, proofEnd, proof, n.leafHashes[proofStart], isMaxNsIgnored), nil
 }
 
+func (n NamespacedMerkleTree) buildRangeProof(proofStart, proofEnd int) [][]byte {
+	proof := [][]byte{}
+	var recurse func(start, end int, includeNode bool) []byte
+	recurse = func(start, end int, includeNode bool) []byte {
+		if start >= len(n.leafHashes) {
+			return nil
+		}
+
+		// reached a leaf
+		if end-start == 1 {
```

Comment:
Did you run gofmt on this file? Doesn't look like we have it in CI, but this line looks sus.

Reply:
Actually, gofmt changes it to this form